chia/util/block_cache.py

```diff
@@ -43,8 +43,16 @@ def height_to_hash(self, height: uint32) -> Optional[bytes32]:
             return None
         return self._height_to_hash[height]
 
-    def contains_block(self, header_hash: bytes32) -> bool:
-        return header_hash in self._block_records
+    def contains_block(self, header_hash: bytes32, height: Optional[uint32] = None) -> bool:
```

Comment:
`height` is still optional here, is that deliberate? I would have thought it's a problem to not implement a compatible protocol.
I suppose this class can't have an underlying blockchain, so we can't have complete knowledge of the main chain, so the behavior of returning false for blocks we don't have in the cache seems reasonable either way. It would probably be good to have a doc-comment about it though

Reply:
fixed